### PR TITLE
feat: Add Options to add Developer Licence to Generated OData Client

### PR DESCRIPTION
**Pull request by FrankEssenberger**
## Context

Added an option for the generator to include a licence file to generated services. I found that the `copyFile()` method was suboptimal in a way that if the read of the source file would take long the callback is potentially never reached because the generator is already finished. A potential solution would be to pass the promise out and wait for them to resolve. However, the current flow was not really written to support this so I decided to make this in a synchronous way.

## Definition of Done

Please consider all items and remove only if not applicable.

- [X] Tests created/adjusted for your changes.
- [X] Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
- [X] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
- [X] If applicable: Properly documented (JSDoc of public API)
- [X] If applicable: Check if `node run doc` still works.

### CODE REVIEW

**Comment by FrankEssenberger** [comment]
I agree, instead of using a flog to say copy File A,B,C... I added now an options `additionalFiles` which takes an glob as an input. All files matching the pattern will be copied to each service.

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,11 +14,13 @@
 
 ## Compatibility Notes
 
--
+- Removed the hidden generator opntion `changelogFile`.
```
Comment (marikaner) [inline CHANGELOG.md:17]: This should be imperative

File: packages/generator/test/generator.spec.ts [hunk 2]
```diff
@@ -1,12 +1,36 @@
...
+
+    expect(
+      project!
+        .getSourceFiles()
+        .filter(
+          file =>
+            file.getBaseName() === 'some-test-markdown.md' ||
+            file.getBaseName() === 'CHANGELOG.md'
+        ).length
+    ).toBe(2);
```
Comment (marikaner) [inline packages/generator/test/generator.spec.ts:31]: This seems like a weird construction, maybe:

```suggestion
    const files = project!.getSourceFiles();
    expect(files.find(file => file.getBaseName() === 'some-test-markdown.md')).toBeDefined();
    expect(files.find(file => file.getBaseName() === 'CHANGELOG.md')).toBeDefined();
```

**Comment by FrankEssenberger** [comment]
> Looks good, I left some comments. I am wondering 2 things though:
> 
> 1. Should we rather deprecate `changelogFile` although is was a hidden option?
> 2. Should we also remove / deprecate the readme option with this?

Regarding 1: I think since the option was marked as hidden it is fair to remove it and use a compability nore.
Regarding : The readme is a file depending on each service so it is not just copied from somewhere [see](https://github.com/SAP/cloud-sdk/blob/1a8c6494ba89eb2dee3e8088d8cdc88d7144183c/packages/generator/src/service/readme.ts#L53-L70) So I guess it is not possible to remove the boolean flag.

File: CHANGELOG.md (hunk 1 above)
Comment (FrankEssenberger) [reply CHANGELOG.md]: adjusted

**Comment by marikaner** [comment]
> > Looks good, I left some comments. I am wondering 2 things though:
> > 
> > 1. Should we rather deprecate `changelogFile` although is was a hidden option?
> > 2. Should we also remove / deprecate the readme option with this?
> 
> Regarding 1: I think since the option was marked as hidden it is fair to remove it and use a compability nore.
> Regarding : The readme is a file depending on each service so it is not just copied from somewhere [see](https://github.com/SAP/cloud-sdk/blob/1a8c6494ba89eb2dee3e8088d8cdc88d7144183c/packages/generator/src/service/readme.ts#L53-L70) So I guess it is not possible to remove the boolean flag.

1: I guess that's fine.
2: I was aware, but now I also got, that this is used by us ;)

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,11 +14,13 @@
...
 
--
+- _OData client generator_: Remove the hidden generator option `changelogFile`.
+Use the new more general options `additionalFiles` instead.
 
 ## New Functionality
 
--
+- Add the option `--additionalFiles <GlobToFiles>` to the generator.
+If specified all files matching the glob will be copied to each generated service directory.
```
Comment (marikaner) [inline CHANGELOG.md:23]:
```suggestion
If specified, all files matching the glob will be copied to each generated service directory.    
```

**Pull request merged**